From 3f8e2016c0411158153b637645dee16be9a3baa9 Mon Sep 17 00:00:00 2001
From: Heiko Schocher <hs@nabladev.com>
Date: Fri, 1 Sep 2023 18:52:48 +0200
Subject: [PATCH 16/23] board: linux: send password if configured

LinuxBootLogin after timeout password was not sent,
when self.password was configured. Sent it!

(May there was crap from login console and prevented
tbot in detecting it)

Signed-off-by: Heiko Schocher <hs@nabladev.com>
---
 tbot/machine/board/linux.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tbot/machine/board/linux.py b/tbot/machine/board/linux.py
index e6da943..c3379ca 100644
--- a/tbot/machine/board/linux.py
+++ b/tbot/machine/board/linux.py
@@ -201,8 +201,9 @@ class LinuxBootLogin(machine.Initializer, LinuxBoot):
                     # should attempt continuing without a password.
                     tbot.log.warning(
                         "Didn't get asked for a password."
-                        + "  Optimistically continuing without one..."
+                        + "  Optimistically send it now..."
                     )
+                    self.ch.sendline(self.password)
                 else:
                     # No timeout exception means we're at the password prompt.
                     self.ch.sendline(self.password)
-- 
2.31.1

